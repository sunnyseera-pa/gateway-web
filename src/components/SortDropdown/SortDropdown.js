import React from 'react';
import { cx } from '@emotion/css';
import Dropdown from '../Dropdown';
import Icon from '../Icon';
import PropTypes from 'prop-types';
import useCommonStyles from '../../hooks/useCommonStyles';
import { PROP_TYPES_DROPDOWN } from '../Dropdown/Dropdown.propTypes';
import { useTranslation } from 'react-i18next';
import { Button } from 'react-bootstrap';

const SortDropdown = ({ onSort, className, options, mt, mb, ml, mr, width, value, direction, allowDirection, ...outerProps }) => {
	const { t } = useTranslation();
	const [state, setState] = React.useState({});
	const commonStyles = useCommonStyles({ mt, mb, ml, mr, width });

	const handleSort = React.useCallback(
		value => {
			const newState = {
				...state,
				value,
			};

			setState(newState);
			onSort(newState);
		},
		[state]
	);

	const handleOrder = React.useCallback(() => {
		const direction = state.direction === 'asc' ? 'desc' : 'asc';
		const newState = {
			...state,
			direction,
		};

		setState(newState);
		onSort(newState);
	}, [state]);

	React.useEffect(() => {
		setState({
			value,
			direction,
		});
	}, [value, direction]);

	return (
		<div className={cx('ui-SortDropdown', className, commonStyles, 'd-flex align-items-center')}>
			<Dropdown
				onSelect={handleSort}
				options={options.map(value => ({
					label: t(`sortby.options.${value}`),
					value,
				}))}
				value={state.value}
				{...outerProps}
			/>
			{allowDirection && (
				<Button onClick={handleOrder} ml={2} variant='link'>
					{state.direction === 'desc' && <Icon name='sort-desc' size='lg' />}
					{state.direction === 'asc' && <Icon name='sort-asc' size='lg' />}
				</Button>
			)}
		</div>
	);
};

SortDropdown.propTypes = {
	...PROP_TYPES_DROPDOWN,
	onSort: PropTypes.func.isRequired,
	allowDirection: PropTypes.bool,
};

SortDropdown.defaultProps = {
	allowDirection: false,
	direction: 'asc',
};

export default SortDropdown;
