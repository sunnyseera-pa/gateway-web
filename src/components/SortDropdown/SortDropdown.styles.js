import { css } from '@emotion/react';

export const root = css`
    display: flex;
    align-items: center;
`;

export const dropdown = css`
    width: 100%;
`;

export const button = css`
    & > span {
        display: flex;
        align-items: center;
    }
`;
