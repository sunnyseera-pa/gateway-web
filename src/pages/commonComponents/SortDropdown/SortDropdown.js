import React, { Suspense } from 'react';
import { Dropdown as BootstrapDropdown } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import Icon from '../Icon';

const SortDropdown = ({ handleSelect, defaultValue, value, type, options, iconSelected }) => {
	const { t } = useTranslation();
	const currentValue = !value ? defaultValue : value;
    console.log(!value);

	return (
        <Suspense fallback={"Loading..."}>
            <p>Test content</p>
            <BootstrapDropdown className='ui-Dropdown' alignRight onSelect={handleSelect}>
                <BootstrapDropdown.Toggle variant='info'>
                    {t(`${type}.label.${currentValue}`)}
                </BootstrapDropdown.Toggle>
                <BootstrapDropdown.Menu>
                    {options.map(item => {
                        return (
                            <BootstrapDropdown.Item eventKey={item} className='gray800-14'>
                                {t(`${type}.option.${item}`)}
                                {item === currentValue && iconSelected}
                            </BootstrapDropdown.Item>
                        );
                    })}
                </BootstrapDropdown.Menu>
            </BootstrapDropdown>
        </Suspense>
	);
};

SortDropdown.defaultProps = {
	options: [],
	defaultValue: 'relevance',
    type: 'sort',
	iconSelected: <Icon name='check' color='green600' ml={1} />,
};

export default SortDropdown;