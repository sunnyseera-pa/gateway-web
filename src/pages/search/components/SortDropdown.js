import React from 'react';
import { Dropdown, Row, Col } from 'react-bootstrap';
import SVGIcon from '../../../images/SVGIcon';
import '../Search.scss';

const SortDropdown = ({ handleSort, sort, dropdownItems, isCollectionsSearch }) => {
	//by default sorted by relevance
	const sorting = sort === '' ? 'relevance' : sort;

	return (
		<Dropdown className='sorting-dropdown' alignRight onSelect={handleSort}>
			<Dropdown.Toggle
				variant='info'
				id='dropdown-menu-align-right'
				className={isCollectionsSearch ? 'collectionsSorting gray800-14' : 'gray800-14'}>
				{(() => {
					if (sorting === 'popularity') return 'Sort by number of views';
					else if (sorting === 'metadata') return 'Sort by metadata quality';
					else if (sorting === 'resources') return 'Sort by related resources';
					else if (sorting === 'latest') return 'Sort by latest';
					else if (sorting === 'recentlyadded') return 'Sort by recently added';
					else if (sorting === 'sortbyyear') return 'Sort by year';
					else if (sorting === 'entries') return 'Number of entries';
					else if (sorting === 'datasets') return 'Number of datasets';
					else return 'Sort by match to search terms';
				})()}
				&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
			</Dropdown.Toggle>
			<Dropdown.Menu>
				{dropdownItems.map((item, index) => {
					return (
						<Row
							key={`ddl-item-${index}`}
							className={
								sorting === item ? 'sort-dropdown-item sort-dropdown-item-selected sortingDropdown' : 'sort-dropdown-item sortingDropdown'
							}>
							<Col xs={12} className='p-0'>
								<Dropdown.Item eventKey={item} className='gray800-14'>
									{(() => {
										if (item === 'popularity') return 'Number of views (highest to lowest)';
										else if (item === 'metadata') return 'Metadata quality (platinum to bronze)';
										else if (item === 'resources') return 'Related resources (most first)';
										else if (item === 'latest') return 'Latest (recently updated first)';
										else if (item === 'recentlyadded') return 'Recently added to collection (newest first)';
										else if (item === 'relevance') return 'Match to search terms (closest first)';
										else if (item === 'sortbyyear') return 'Sort by year (latest first)';
										else if (item === 'entries') return 'Number of entries (most first)';
										else if (item === 'datasets') return 'Number of datasets (most first)';
									})()}
								</Dropdown.Item>
							</Col>
							<div className='p-0 sortingCheckmark'>
								{sorting === item ? (
									<SVGIcon
										name='check'
										width={20}
										height={20}
										visble='true'
										style={{
											float: 'right',
											fill: '#3db28c',
											marginTop: '5px',
										}}
										fill={'#3db28c'}
										stroke='none'
									/>
								) : null}
							</div>
						</Row>
					);
				})}
			</Dropdown.Menu>
		</Dropdown>
	);
};

export default SortDropdown;
