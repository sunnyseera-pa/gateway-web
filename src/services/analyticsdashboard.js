import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
