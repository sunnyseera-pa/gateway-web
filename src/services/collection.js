import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
