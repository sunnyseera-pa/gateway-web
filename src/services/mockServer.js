import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';
import mswSearch from './search/mockMsw';
import mswGetLocations from './locations/mockMsw';
import mswGetContributors from './contributors/mockMsw';

import translations from '../../public/locales/en-GB/translation.json';

const mswGetEnTranslations = rest.get(`http://localhost/locales/en/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetEnGbTranslations = rest.get(`http://localhost/locales/en-GB/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetIcon = rest.get(`/images/Application_approved.svg`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.text('<svg />'));
});

const handlers = [
	...mswDatasets,
	...mswDatasetOnboarding,
	...mswPostDatasetActivityLog,
	// ...mswSearch,
	...mswGetLocations,
	...mswGetContributors,
	mswGetEnTranslations,
	mswGetEnGbTranslations,
	mswGetIcon,
];

export const server = setupServer(...handlers);
