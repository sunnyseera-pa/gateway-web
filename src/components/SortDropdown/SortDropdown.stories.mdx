import { Meta, Story, Canvas } from '@storybook/addon-docs';

import SortDropdown from './SortDropdown';

<Meta
	title='Core/SortDropdown'
	component={SortDropdown}
	args={{
		options: ['recentlyadded', 'relevance'],
	}}
/>

<Canvas>
	<Story name='Basic'>
		{props => {
			const [value, setValue] = React.useState('relevance');
			const handleSort = React.useCallback(value => {
				setValue(value);
			}, []);
			return <SortDropdown {...props} onSort={handleSort} value={value} width='350px' />;
		}}
	</Story>
</Canvas>
