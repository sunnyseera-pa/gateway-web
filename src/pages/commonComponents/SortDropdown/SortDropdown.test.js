import { render } from '@testing-library/react';
import React from 'react';
import SortDropdown from '.';
import { server } from '../../../services/mockServer';

let wrapper;

const props = {
    handleSelect: jest.fn(),
    defaultValue: 'recentActivity',
    value: 'recentlyPublished',
    type: 'sort',
    options: [
        'popularity',
        'metadata',
		'resources',
		'latest',
		'recentlyadded',
        'sortbyyear',
        'relevance'
    ]
}

describe('Given the Dropdown component', () => {
	describe('When it is rendered', () => {
		beforeAll(() => {
			server.listen();
			wrapper = render(<SortDropdown {...props} />, {
				wrapper: Providers,
			});
		});

		afterEach(() => {
            server.resetHandlers();
        });

		afterAll(() => {
            server.close();
		});

		it('Then matches the previous snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});
	});
});
