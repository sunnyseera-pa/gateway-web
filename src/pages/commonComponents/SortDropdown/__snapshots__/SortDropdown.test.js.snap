// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Dropdown component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <p>
    Test content
  </p>
  <div
    class="ui-Dropdown dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle btn btn-info"
      type="button"
    >
      sort.label.recentlyPublished
    </button>
  </div>
</div>
`;
